src/bench/connectblock.cpp

```diff
@@ -0,0 +1,133 @@
+// Copyright (c) 2025 The Bitcoin Core developers
+// Distributed under the MIT software license, see the accompanying
+// file COPYING or http://www.opensource.org/licenses/mit-license.php.
+
+#include <addresstype.h>
+#include <bench/bench.h>
+#include <interfaces/chain.h>
+#include <kernel/cs_main.h>
+#include <script/interpreter.h>
+#include <sync.h>
+#include <test/util/setup_common.h>
+#include <validation.h>
+
+#include <cassert>
+#include <vector>
+
+CBlock CreateTestBlock(
+    std::shared_ptr<TestChain100Setup> test_setup,
+    const std::vector<CKey>& keys,
+    const std::vector<CScript>& scriptpubkeys,
+    int num_txs = 1000)
+{
+    Chainstate& chainstate{test_setup->m_node.chainman->ActiveChainstate()};
+
+    std::vector<CTxOut> outputs;
+    // Each transaction will pay to the same outputs that will be spent in the next one
+    outputs.reserve(scriptpubkeys.size());
+    for (const auto& spk : scriptpubkeys) {
+        outputs.emplace_back(COIN, spk);
+    }
```

Comment:
It seems this was moved in here to deduplicate the code (which is great!), but I found this part to be a bit confusing. We only use this `scriptpubkeys` argument to create an outputs array, so why not create the `CTxOut` vector directly when we are creating the `scriptpubkeys` vector and then pass the `CTxOut` vector in as an argument? I also think a comment explicitly mentioning we are creating an same number input, same number output transaction would be worth adding.

Reply:
> so why not create the `CTxOut` vector directly when we are creating the `scriptpubkeys` vector and then pass the `CTxOut` vector in as an argument?

I updated the PR to do that.

> I also think a comment explicitly mentioning we are creating an same number input, same number output transaction would be worth adding.

I also added a comment stating this